types/mui-datatables/index.d.ts

```diff
@@ -290,18 +300,22 @@ export type MUIDataTableOptions = Partial<{
         columns: any,
         data: any,
     ) => string | boolean;
-    onFilterChange: (changedColumn: string, filterList: any[], type: FilterType | 'chip' | 'reset') => void;
+    onFilterChange: (
+        changedColumn: string | MUIDataTableColumn | null,
```

Comment:
- `null` [when resetting filters](https://github.com/gregnb/mui-datatables/blob/master/src/MUIDataTable.js#L1372)
- If a column has `filterType: 'custom'` it is passed a `MUIDataTableColumn`
- Lastly, a regular column is passed a string of the column name